app/redux/thunks.js

```diff
@@ -0,0 +1,154 @@
+import axios from 'axios';
+
+import {
+    setProducts,
+    setSingleProduct,
+    setUser,
+    setOrders,
+} from 'actions.js';
+
+//Thunk for fetching all products. 
+//Do you think we could initialize id to null, and reuse this thunk to make a call for a single product?
+export const fetchProducts = () => {
+    return dispatch => {
+        return axios.get(`/api/products`)
+        .then(res => dispatch(setProducts(res.data)))
+        .catch(e => console.error('Error fetching products', e))
+    }
+}
+
+//Thunk for fetching a single product
+export const fetchSingleProduct = productId => {
+    return dispatch => {
+        return axios.get(`/api/products/${ productId }`)
+        .then(res => dispatch(setSingleProduct(res.data)))
+        .catch(e => console.error('Error fetching single product', e))
+    }
+}
+
+//Thunk for creating a product.
+//Refetches products after creating.
+export const postProduct = product => {
+    return dispatch => {
+        return axios.post(`/api/products`, product)
+        .then(() => dispatch(fetchProducts()))
+        .catch(e => console.error('Error creating product', e))
+    }
+}
+
+//Thunk for deleting a product.
+//Refetches products after deleting.
+export const deleteProduct = productId => {
+    return dispatch => {
+        return axios.delete(`/api/products/${ productId }`)
+        .then(() => dispatch(fetchProducts()))
+        .catch(e => console.error('Error deleting product', e))
+    }
+}
+
+//Thunk for updating a product.
+//Refetches the product after updating.
+export const updateProduct = (productId, product) => {
+    return dispatch => {
+        return axios.put(`/api/products/${ productId }`, product)
+        .then(res => dispatch(setSingleProduct(res.data)))
+        .catch(e => console.error('Error updating product', e))
+    }
+}
+
+//Thunk for fetching a user
+export const fetchUser = (userId = null) => {
+    return dispatch => {
+        return axios.get(`/api/users/${ userId }`)
+        .then(res => dispatch(setUser(res.data)))
+        .catch(e => console.error('Error fetching user', e))
```

Comment:
Good point about error handling. If the error occurs during a read or write action to the server, then we may want to create an "error component" that displays when there is a critical error.